input-explainer.md

```diff
@@ -27,15 +27,15 @@ The properties of an XRInputSource object are immutable. If a device can be mani
 
 Each input source can query a `XRInputPose` using the `getInputPose()` function of any `XRFrame`. Getting the pose requires passing in the `XRInputSource` you want the pose for, as well as the `XRReferenceSpace` the pose values should be given in, just like `getViewerPose()`. `getInputPose()` may return `null` in cases where tracking has been lost (similar to `getViewerPose()`), or the given `XRInputSource` instance is no longer connected or available.
 
-The `gripMatrix` is a transform into a space where if the user was holding a straight rod in their hand it would be aligned with the negative Z axis (forward) and the origin rests at their palm. This enables developers to properly render a virtual object held in the user's hand. For example, a sword would be positioned so that the blade points directly down the negative Z axis and the center of the handle is at the origin.
+The `grip` is an `XRRigidTransform` into a space where if the user was holding a straight rod in their hand it would be aligned with the negative Z axis (forward) and the origin rests at their palm. This enables developers to properly render a virtual object held in the user's hand. For example, a sword would be positioned so that the blade points directly down the negative Z axis and the center of the handle is at the origin.
 
-If the input source has only 3DOF, the grip matrix may represent only a translation or rotation based on tracking capability. An example of this case is for physical hands on some AR devices which only have a tracked position. The `gripMatrix` will be `null` if the input source isn't trackable. 
+If the input source has only 3DOF, the grip transform may represent only a translation or rotation based on tracking capability. An example of this case is for physical hands on some AR devices which only have a tracked position. The `grip` will be `null` if the input source isn't trackable. 
```

Comment:
Any given tracked entity (both viewers and input sources) can also _temporarily_ lose its 6DOF position.  For example, an inside-out tracked motion controller can move outside of the headset's tracking volume, but can still use its IMU to provide orientation updates.

For any given `XRRigidTransform`, we'll likely need to indicate if the position is a true detected position or some sort of "last known" position.  We could do this as either a boolean `positionTracked` attribute or by making `position` itself nullable - the former feels more like a "pit of success" for developers who don't consider such edge cases.

Reply:
`XRInputPose.emulatedPosition` could handle this, but then we need an equivalent attribute on `XRViewerPose`.  Apps can also generically get into precisely the same situation with any relation between two `XRSpace`s.  It may make sense to just promote `emulatedPosition` onto `XRRigidTransform` in general, now that we have a richer transform type that could hold it.

Reply:
Trying to give a timestamp may expose more nuance than we want to tackle right now, however.  For example:
* There may be a timestamp for the last sensor reading, but then the position is forward-predicted to this `XRFrame`'s photon time.  In that case, is the timestamp the timestamp of the last actual reading or the target timestamp of the prediction?
* If an inside-out-tracked motion controller stops being observed long enough (e.g. is behind your back), it might stop getting logical controller-to-head position updates, and so the developer would expect to check that by observing the timestamp is fixed from that point forward.  However, when asking where the controller is relative to the _world_, the UA should continue to incorporate updated head-to-world position updates, which means that the transform would continue to update every frame, even though it is now logically "estimated".  Would an app that gets back the same old timestamp again expect that it also gets precisely the same value for its transform?

For WebXR 1.0, the best balance here may just be a bool with some name like "estimatedPosition", "approximatePosition" or "inferredPosition".

Reply:
This is a really good line of discussion and I agree we should address it, but I've actually been working on something that feels like it would give us a cleaner place to put the `estimated` flags that I want to propose in this repo Real Soon Now™️, so I'd prefer to land this PR first and then work on this specific problem (which fortunately should be an additive thing either way.)

Filed issue #445 to ensure it gets tracked.

Reply:
Sounds good to me.